CMakeLists.txt

```diff
@@ -137,15 +137,18 @@ find_package(ZLIB REQUIRED)
 find_package(Threads REQUIRED)
 
 option(ENABLE_PLUGINS "Compile with plugin support" ON)
-find_package(Lua 5.3) # Lua 5.4 is only supported with cmake >=3.18
+find_package(Lua) # Lua 5.4 is only supported with cmake >=3.18
+message(STATUS "Found Lua: ${Lua_FOUND}")
 if (Lua_FOUND AND ENABLE_PLUGINS)
     # currently not fully supported by cmake
-    message("Enable Xournal++ Plugins")
+    message(STATUS "Enable Xournal++ Plugins")
     add_library(lua INTERFACE)
     target_link_libraries(lua INTERFACE ${LUA_LIBRARIES})
     target_include_directories(lua INTERFACE ${LUA_INCLUDE_DIR})
+    target_compile_definitions(lua INTERFACE ENABLE_PLUGINS)
```

Comment:
@Febbe Do you remember why you added this line here? It causes some
```
./build/src/config-features.h:20: warning: "ENABLE_PLUGINS" redefined
   20 | #define ENABLE_PLUGINS
```
all over the place when compiling with GCC.

We can always remove the line from config-features.h (it works), but it sounds dirtier than to remove the above line.

Reply:
I've re-added it after someone removed it, because I got compilation errors.
It's probably some sort of religion/opinion: My opinion is, that those variables should be configured via CMake, so that they are available in all relevant source and header files, this reduces bugs, when a file forgets to include the config file.
It's probably the more modern way.

But of course it has drawbacks: It can happen, that this way might affect included third party headers, like the Windows API, when a name clash occurs. Including only the config file, where it is required, can circumvent this.
But that's a design flaw anyway, header files should always work the same regardless of the order in which they are included.
For example, the llvm include order enforces this, by including local headers first and system headers last. We use the google include order, which relaxes exactly this to retain some compatibility to old and badly designed C libraries. It is also more readable.

You can remove one of them, which one might not matter, your solution looks cleaner, because it's more consistent. But I think it would be cleaner, to configure all options in CMake. I also think `configure_file` has a big problem:

It only reconfigures, when the source.in file changed, not any of the variables:

> If the input file is modified the build system will re-run CMake to re-configure the file and generate the build system again. The generated file is modified and its timestamp updated on subsequent cmake runs only if its content is changed.
 - https://cmake.org/cmake/help/latest/command/configure_file.html